src/app/new-auth/components/input-field/input-field.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, Input, OnInit, forwardRef } from '@angular/core';
+import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { AuthValidationService } from 'app/core/auth-validation.service';
+/**
+ * input field for authentication module using
+ * angular material input field and designed
+ * to be used with ngModel
+ *
+ * @param pwrd: Boolean - make true if the field is for a
+ * password. default 'false'
+ * @param phold: String - placeholder text, which becomes label
+ * @param fControlType: 'email' | 'username' | 'password' | 'text' -
+ * describes the type of validation needed for the field. default 'text'
+ * no validation
+ */
+@Component({
+  selector: 'clark-input-field',
+  templateUrl: './input-field.component.html',
+  styleUrls: ['./input-field.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(
+        () => InputFieldComponent
+      ),
+      multi: true
+    }
+  ]
+})
+export class InputFieldComponent implements OnInit, ControlValueAccessor {
+  @Input() pwrd: Boolean = false;
+  @Input() phold: String = '';
+  @Input() fControlType: 'email' | 'userName' | 'password' | 'text' = 'text';
```

Comment:
= 'text' is the default value for fControlType. 'email' | 'username' | 'password' | 'text' are the strings that it will accept (type). it does look weird tho